SimulationAgent/SimulationRunner.cs

```diff
@@ -127,6 +126,24 @@ public void Start(Services.Models.Simulation simulation)
                 // the active hub, e.g. in case the user provided a custom connection string.
                 this.devices.SetCurrentIotHub();
 
+                // Create the devices
+                try
+                {
+                    var devices = this.simulations.GetDeviceIds(simulation);
+
+                    // This will ignore existing devices, creating only the missing ones
+                    this.devices.CreateListAsync(devices).Wait(TimeSpan.FromSeconds(30));
+                }
+                catch (Exception e)
+                {
+                    this.running = false;
+                    this.starting = false;
+                    this.log.Error("Failed to create devices", () => new { e });
+
```

Comment:
If creation fails, do we delete the devices that may have been created on the hub?

Reply:
Not at this stage, we only care that devices exists. (this was the case also before, also in RM)